specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2220 @@
+          "default": false,
+          "type": "boolean"
+        }
+      }
+    },
+    "Build": {
+      "description": "The properties for a build.",
+      "type": "object",
+      "properties": {
+        "buildId": {
+          "description": "The unique identifier for the build.",
+          "type": "string"
+        },
+        "status": {
+          "description": "The current status of the build.",
+          "enum": [
+            "Queued",
+            "Started",
+            "Running",
+            "Succeeded",
+            "Failed",
+            "Cancelled"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "BuildStatus",
+            "modelAsString": true
+          }
+        },
+        "lastUpdatedTime": {
+          "format": "date-time",
+          "description": "The last updated time for the build.",
+          "type": "string"
+        },
+        "buildType": {
+          "description": "The type of build.",
+          "enum": [
+            "AutoBuild",
+            "QuickBuild"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "BuildType",
+            "modelAsString": true
+          }
+        },
+        "createTime": {
+          "format": "date-time",
+          "description": "The time the build was created.",
+          "type": "string"
+        },
+        "startTime": {
+          "format": "date-time",
+          "description": "The time the build started.",
+          "type": "string"
+        },
+        "finishTime": {
+          "format": "date-time",
+          "description": "The time the build finished.",
+          "type": "string"
+        },
+        "outputImages": {
+          "description": "The list of all images that were generated from the build.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ImageDescriptor"
+          }
+        },
+        "buildDefinition": {
+          "description": "All the properties of the build definition with which the build was started.",
+          "type": "string"
+        },
+        "trigger": {
+          "description": "The trigger that caused the build.",
+          "type": "string"
+        },
+        "isArchiveEnabled": {
+          "description": "The value that indicates whether archiving is enabled or not.",
+          "type": "boolean"
+        },
+        "platform": {
+          "$ref": "#/definitions/PlatformProperties",
+          "description": "The platform properties against which the build will happen."
+        }
+      }
+    },
+    "ImageDescriptor": {
+      "description": "Properties for a registry image.",
+      "required": [
+        "imageName",
+        "digest"
+      ],
+      "type": "object",
+      "properties": {
+        "imageName": {
+          "description": "Name of the repository or image.",
+          "type": "string"
+        },
+        "tag": {
```

Comment:
Isn't `tag` part of `imageName`?

Reply:
We have been interchangeably using image name to repository. In all our case, we separated the repository to tag. Do you suggest to rename to repository?

Reply:
I think we should either (1) only have `imageName` and use it consistently with `docker`. For example, `ubuntu:16.04` and `ubuntu` just means `ubuntu:latest`; (2) or have `repository` and `tag` as two different fields. Az CLI uses the latter pattern.